scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtConfig.scala

```diff
@@ -140,19 +140,21 @@ case class ScalafmtConfig(
     fileOverride: Conf.Obj = Conf.Obj.empty,
     xmlLiterals: XmlLiterals = XmlLiterals()
 ) {
-  lazy val alignMap: Map[String, Seq[AlignToken.Matcher]] =
+  private[scalafmt] lazy val alignMap: Map[String, Seq[AlignToken.Matcher]] =
     align.tokens.map(x => x.code -> x).toMap.map { case (k, v) =>
       k -> v.getMatcher
     }
 
-  def withDialect(dialect: ScalafmtRunner.Dialect.WithName): ScalafmtConfig =
+  private[scalafmt] def withNamedDialect(
+      dialect: ScalafmtRunner.Dialect.WithName
+  ): ScalafmtConfig =
     copy(runner = runner.copy(dialect = dialect))
 
-  def withDialect(
```

Comment:
the problem reported is due to us adding a default parameter to this method, so splitting into two -- that's the primary fix.